test/intl402/DisplayNames/prototype/of/type-language-invalid.js

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2023 Igalia S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-Intl.DisplayNames.prototype.of
+description: Throws a RangeError for invalid `language` codes
+info: |
+  12.3.3 Intl.DisplayNames.prototype.of ( code )
+
+  1. If type is "language", then
+      a. If code cannot be matched by the unicode_language_id Unicode locale nonterminal, throw a RangeError exception.
+      b. If IsStructurallyValidLanguageTag(code) is false, throw a RangeError exception.
+      c. Return CanonicalizeUnicodeLocaleId(code).
+features: [Intl.DisplayNames-v2]
+---*/
+
+var displayNames = new Intl.DisplayNames(undefined, {type: 'language'});
+
+assert.throws(RangeError, function() {
+  displayNames.of('');
+}, 'invalid language subtag - empty string');
+
+assert.throws(RangeError, function() {
+  displayNames.of('a');
+}, 'invalid language subtag - only one character');
+
+assert.throws(RangeError, function() {
+  displayNames.of('abcdefghi');
+}, 'invalid language subtag - > 8 characters');
+
+assert.throws(RangeError, function() {
+  displayNames.of('en-u-hebrew');
+}, 'invalid language id -- contains singleton');
+
+assert.throws(RangeError, function() {
+  displayNames.of('enxx');
+}, 'invalid language subtag (4 letters)');
```

Comment:
```suggestion
assert.throws(RangeError, function() {
  displayNames.of('enxx');
}, 'invalid language id - starts with a script subtag (4 letters)');
```

But actually, how does this differ from "tries to use backwards compatibility feature (bare script subtag)" below?

Reply:
oh, fair! let me take that one out...